File: Day 3/lcm.c
#include <stdio.h>

int main() {
    int n1, n2, temp1, temp2, rem, gcd, lcm;

    printf("Enter first number: ");
    scanf("%d", &n1);
    printf("Enter second number: ");
    scanf("%d", &n2);

    temp1 = n1;
    temp2 = n2;

    // Step 1: Find the GCD first
    while (temp2 != 0) {
        rem = temp1 % temp2;
        temp1 = temp2;
        temp2 = rem;
    }
    gcd = temp1;

    // Step 2: Use the standard mathematical formula: LCM = (n1 * n2) / GCD
    lcm = (n1 * n2) / gcd;

    printf("LCM of %d and %d is: %d\n", n1, n2, lcm);

    return 0;
}

File: Day 3/lcm.java
import java.util.Scanner;

public class lcm {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter first number: ");
        int n1 = sc.nextInt();
        System.out.print("Enter second number: ");
        int n2 = sc.nextInt();
        
        int temp1 = Math.abs(n1);
        int temp2 = Math.abs(n2);
        
        // Find GCD first
        while (temp2 != 0) {
            int rem = temp1 % temp2;
            temp1 = temp2;
            temp2 = rem;
        }
        int gcd = temp1;
        
        // LCM formula: (a * b) / GCD
        int lcm = (Math.abs(n1) * Math.abs(n2)) / gcd;
        
        System.out.println("LCM of " + n1 + " and " + n2 + " is: " + lcm);
        sc.close();
    }
}